### PR TITLE
Support 6/7 mp-rest properties

### PR DESCRIPTION
**Pull request by angelozerr**
Support 6/7 mp-rest properties

See https://github.com/redhat-developer/quarkus-ls/issues/189

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java [hunk 1]
```diff
@@ -131,6 +138,20 @@ protected void processAnnotation(IJavaElement javaElement, IAnnotation registerR
 				super.addItemMetadata(collector, MP_REST_CLASS_REFERENCE_TYPE + "/mp-rest/providers",
 						"java.lang.String", docs, null, null, null, null, null, false);
 
+				// TODO :
+				// com.mycompany.remoteServices.MyServiceClient/mprest/providers/com.mycompany.MyProvider/priority
+				// will override the priority of the provider for this interface.
+
+				// /mp-rest/connectTimeout
+				docs = "Timeout specified in milliseconds to wait to connect to the remote endpoint.";
+				super.addItemMetadata(collector, MP_REST_CLASS_REFERENCE_TYPE + "/mp-rest/connectTimeout",
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: there's no validation error if the value is not a number

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java [hunk 2]
```diff
@@ -131,6 +138,20 @@ protected void processAnnotation(IJavaElement javaElement, IAnnotation registerR
...
 						"java.lang.String", docs, null, null, null, null, null, false);
 
+				// TODO :
+				// com.mycompany.remoteServices.MyServiceClient/mprest/providers/com.mycompany.MyProvider/priority
+				// will override the priority of the provider for this interface.
+
+				// /mp-rest/connectTimeout
+				docs = "Timeout specified in milliseconds to wait to connect to the remote endpoint.";
+				super.addItemMetadata(collector, MP_REST_CLASS_REFERENCE_TYPE + "/mp-rest/connectTimeout",
+						"java.lang.Integer", docs, null, null, null, null, null, false);
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: not Long?

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java [hunk 3]
```diff
@@ -131,6 +138,20 @@ protected void processAnnotation(IJavaElement javaElement, IAnnotation registerR
...
+
+				// /mp-rest/readTimeout
+				docs = "Timeout specified in milliseconds to wait for a response from the remote endpoint.";
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: there's no validation error if the value is not a number

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java [hunk 4]
```diff
@@ -131,6 +138,20 @@ protected void processAnnotation(IJavaElement javaElement, IAnnotation registerR
...
+				super.addItemMetadata(collector, MP_REST_CLASS_REFERENCE_TYPE + "/mp-rest/readTimeout",
+						"java.lang.Integer", docs, null, null, null, null, null, false);
```
Comment (fbricon) [inline microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: Long?

**Comment by fbricon** [comment]
Travis build failed:
```
Failures: 
  MicroProfileRegisterRestClientTest.restClientQuickstart:40 ${mp.register.rest.client.class}/mp-rest/conectionTimeout should only exist once: Actual: ${mp.register.rest.client.class}/mp-rest/url,${mp.register.rest.client.class}/mp-rest/uri,${mp.register.rest.client.class}/mp-rest/scope,${mp.register.rest.client.class}/mp-rest/providers,${mp.register.rest.client.class}/mp-rest/connectTimeout,${mp.register.rest.client.class}/mp-rest/readTimeout expected:<1> but was:<0>
```

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java (hunk 1 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: Good catch. There were a bug while expend of properties. I fixed it and write test for that.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java (hunk 2 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: Right, fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java (hunk 3 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: should be fixed.

File: microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java (hunk 4 above)
Comment (angelozerr) [reply microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java]: should be fixed.

**Comment by angelozerr** [comment]
> Travis build failed:

Should be fixed.

**Pull request merged**